src/psyclone/psyir/nodes/omp_task_directive.py

```diff
@@ -0,0 +1,2136 @@
+        # we need remove the children from that node before adding
+        # the children from this node.
+        if lowering:
+            sched_childs = children[0].pop_all_children()
+            super().__init__(
+                children=sched_childs, parent=parent
+            )
+            for child in children[1:]:
+                self.addchild(child)
+        else:
+            super().__init__(children=children, parent=parent)
+
+    @staticmethod
+    def _validate_child(position, child):
+        """
+         Decides whether a given child and position are valid for this node.
+         The rules are:
+         1. Child 0 must always be a Schedule.
+         2. Child 1 must always be an OMPPrivateClause
+         3. Child 2 must always be an OMPFirstprivateClause
+         4. Child 3 must always be an OMPSharedClause
+         5. Child 4 and 5 must always be OMPDependClauses
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        """
+        if position == 0:
+            return isinstance(child, Schedule)
+        if position == 1:
+            return isinstance(child, OMPPrivateClause)
+        if position == 2:
+            return isinstance(child, OMPFirstprivateClause)
+        if position == 3:
+            return isinstance(child, OMPSharedClause)
+        if position in (4, 5):
+            return isinstance(child, OMPDependClause)
+        return False
+
+    @property
+    def input_depend_clause(self):
+        """
+        :returns: the OMPDependClause child of this node corresponding to \
+                  input dependencies.
+        :rtype: :py:class:`psyclones.psyir.nodes.OMPDependClause`
+        """
+        return self.children[4]
+
+    @property
+    def output_depend_clause(self):
+        """
+        :returns: the OMPDependClause child of this node corresponding to \
+                  output dependencies.
+        :rtype: :py:class:`psyclones.psyir.nodes.OMPDependClause`
+        """
+        return self.children[5]
+
+    def begin_string(self):
+        """Returns the beginning statement of this directive, i.e.
+        "omp task ...". The visitor is responsible for adding the
+        correct directive beginning (e.g. "!$").
+
+        :returns: the beginning statement for this directive.
+        :rtype: str
+
+        """
+        # Generate the string containing the required clauses
+        return "omp task"
+
+    def end_string(self):
+        """Returns the end (or closing) statement of this directive, i.e.
+        "omp end task". The visitor is responsible for adding the
+        correct directive beginning (e.g. "!$").
+
+        :returns: the end statement for this directive.
+        :rtype: str
+
+        """
+        # pylint: disable=no-self-use
```

Comment:
My pylint says this statement is 'useless' (harsh).

Reply:
Yeah, removing it doesn't seem to result in any complaints.